File: lang/php/test/vac.fun/algorithms/sorting/SelectionSortTest.php
<?php
    require_once "vac.fun/algorithms/sorting/SelectionSort.php";
    require_once "vac.fun/util/SequenceBuilder.php";
    require_once "vac.fun/util/Sequences.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\algorithms\sorting\SelectionSort;
    use vac\fun\util\SequenceBuilder;
    use vac\fun\util\Sequences;
    use vac\fun\util\TestRunner;

    function testSelectionSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        SelectionSort\sort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testSelectionSort());
    }
?>

File: lang/ruby/test/vac.fun/algorithms/sorting/SelectionSortTest.rb
require "vac.fun/algorithms/sorting/SelectionSort"
require "vac.fun/util/SequenceBuilder"
require "vac.fun/util/Sequences"
require "vac.fun/util/TestRunner"

def testSelectionSort()
    size = 32768

    arr = Array.new(size)
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    SelectionSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum
        return false
    end

    Sequences.isSorted(arr)
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testSelectionSort())
end
